python/src/functions/create_archive.py

```diff
@@ -0,0 +1,76 @@
+import json
+import os
+import tempfile
+import zipfile
+
+import boto3
+from aws_lambda_typing.context import Context
+from aws_lambda_typing.events import SQSEvent
+from mypy_boto3_s3.client import S3Client
+from src.lib.logging import get_logger, reset_contextvars
+
+S3_BUCKET = os.environ.get("S3_BUCKET", "test_bucket")
+
+
+@reset_contextvars
+def handle(event: SQSEvent, _context: Context):
```

Comment:
As discussed on the Friday call the main change here is to update the input event to be a step-function event rather than an `SQSEvent`.